### PR TITLE
add support for ACF Blocks for Gutenberg

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 1]
```diff
@@ -268,6 +269,10 @@ public function applyXML(SubmissionEntity $submission, $xml)
             do_action(ExportedAPI::EVENT_SMARTLING_AFTER_DESERIALIZE_CONTENT, $params);
             if (array_key_exists('entity', $translation) && ArrayHelper::notEmpty($translation['entity'])) {
                 $translation['entity'] = self::arrayMergeIfKeyNotExists($lockedData['entity'], $translation['entity']);
+                $acf = AcfDynamicSupport::getAcf();
+                if ($acf !== null && version_compare($acf->version, '5.8.0') >= 0) {
```
Comment (dimitrystd) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php:273]: It seems it will stop to work for 5.8.1 and other versions

File: readme.txt [hunk 2]
```diff
@@ -58,6 +58,9 @@ Additional information on the Smartling Connector for WordPress can be found [he
 3. Track translation status within WordPress from the Submissions Board. View overall progress of submitted translation requests as well as resend updated content.
 
 == Changelog ==
+= 1.12.6 =
+Add support for ACF Blocks for Gutenberg
```
Comment (dimitrystd) [inline readme.txt:62]:
```suggestion
Add translation support for ACF Blocks in Gutenberg UI
```
Didn't get the trick with ACF version. So potentially we should mention it here too.

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php (hunk 1 above)
Comment (PavelLoparev) [reply inc/Smartling/Base/SmartlingCoreUploadTrait.php:273]: Didn't get why. `version_compare` returns 0 if equals and 1 if arg1 > arg2.

File: inc/Smartling/Extensions/Acf/AcfDynamicSupport.php [hunk 3]
```diff
@@ -76,18 +73,29 @@ public function getSiteHelper()
...
+     * @return string
+     */
+    public static function prepareGutenbergBlocks($text) {
+        return preg_replace_callback(
+            '/<!--\s+wp:(?P<name>[\S]+)\s+(?P<attrs>{[\S\s]+?})\s+(?P<void>\/)?-->/',
+            function (array $matches) {
+                return addslashes($matches[0]);
+            },
+            $text
+        );
```
Comment (PavelLoparev) [inline inc/Smartling/Extensions/Acf/AcfDynamicSupport.php:98]: Could you please provide an example like "before/after" for this function?

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php (hunk 1 above)
Comment (PavelLoparev) [reply inc/Smartling/Base/SmartlingCoreUploadTrait.php:273]: @vsolovei-smartling what is changed in ACF 5.8.0?

Comment (vsolovei-smartling) [reply inc/Smartling/Base/SmartlingCoreUploadTrait.php:273]: https://www.advancedcustomfields.com/blog/acf-5-8-introducing-acf-blocks-for-gutenberg/

Comment (vsolovei-smartling) [reply inc/Smartling/Base/SmartlingCoreUploadTrait.php:273]: It works with version 5.8.0 and after, I've tested locally with plugin versions 5.8.7 and 5.8.9

File: inc/Smartling/Extensions/Acf/AcfDynamicSupport.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Extensions/Acf/AcfDynamicSupport.php]: Reworked the fix, made it depend on acf functions, tried adding mocks to test, that doesn't quite work.

File: inc/Smartling/Helpers/GutenbergBlockHelper.php [hunk 4]
```diff
@@ -305,9 +305,15 @@ public function renderGutenbergBlock($name, array $attrs = [], array $chunks = [
...
-        return ('' !== $content)
+        $result = ('' !== $content)
             ? vsprintf('<!-- wp:%s%s -->%s<!-- /wp:%s -->', [$name, $attributes, $content, $name])
             : vsprintf('<!-- wp:%s%s /-->', [$name, $attributes]);
+
+        if ((function_exists('acf_parse_save_blocks') && function_exists('acf_has_block_type')) && acf_has_block_type($name)) {
+            $result = addslashes($result);
+        }
+
+        return $result;
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/GutenbergBlockHelper.php:316]: In this way you check that this is new acf plugin?

Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/GutenbergBlockHelper.php:316]: Yes, acf_parse_save_blocks was added at version 5.7.13 of ACF, and is the one that breaks json in acf.
acf_has_block_type ensures that a block belongs to acf and needs to be slashed.

**Pull request merged**